app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list[dict]) -> list[Person]:
+    person_list = [
+        Person(name=person_data["name"], age=person_data["age"])
+        for person_data in people
+    ]
+
+    for person_data in people:
+        person = Person.people[person_data["name"]]
+
+        if person_data.get("wife"):
+            person.wife = Person.people[person_data["wife"]]
+        if person_data.get("husband"):
+            person.husband = Person.people[person_data["husband"]]
+
+    return person_list
```

Comment:
The `create_person_list` function seems to do more than just creating a list of `Person` objects. It also assigns husbands and wives to `Person` objects. It's a good practice to make sure that a function does only one thing (Single Responsibility Principle). Consider splitting this function into two: one for creating `Person` objects and another for assigning husbands and wives.